packages/components/countdown/src/countdown.vue

```diff
@@ -0,0 +1,80 @@
+<template>
+  <el-statistic
+    :value="rawValue"
+    :title="title"
+    :prefix="prefix"
+    :suffix="suffix"
+    :value-style="valueStyle"
+    :formatter="formatter"
+  >
+    <template v-for="(_, name) in $slots" #[name]>
+      <slot :name="name" />
+    </template>
+  </el-statistic>
+</template>
+<script lang="ts" setup>
+import { computed, onBeforeUnmount, ref, watch } from 'vue'
+import { ElStatistic } from '@element-plus/components/statistic'
+import { countdownEmits, countdownProps } from './countdown'
+import { formatTime, getTime } from './utils'
+
+defineOptions({
+  name: 'ElCountdown',
+})
+
+const REFRESH_INTERVAL = 1000 / 30
+const props = defineProps(countdownProps)
+const emit = defineEmits(countdownEmits)
+
+let timer: ReturnType<typeof setInterval> | undefined
+
+const rawValue = ref(getTime(props.value) - Date.now())
+
+const displayValue = computed(() => formatTime(rawValue.value, props.format))
+
+const formatter = (val: number) => formatTime(val, props.format)
+
+const stopTimer = () => {
+  if (timer) {
+    clearInterval(timer)
+    timer = undefined
+  }
+}
+
+const startTimer = () => {
+  const timestamp = getTime(props.value)
+
+  timer = setInterval(() => {
+    let diff = timestamp - Date.now()
+    emit('change', diff)
+    if (diff <= 0) {
+      diff = 0
+      stopTimer()
+      emit('finish')
+    }
+    rawValue.value = diff
+  }, REFRESH_INTERVAL)
+}
```

Comment:
```suggestion
import { computed, onBeforeUnmount, ref, watch } from 'vue'
import { ElStatistic } from '@element-plus/components/statistic'
import { cAF, rAF } from '@element-plus/utils/raf'
import { countdownEmits, countdownProps } from './countdown'
import { formatTime, getTime } from './utils'

defineOptions({
  name: 'ElCountdown',
})

const props = defineProps(countdownProps)
const emit = defineEmits(countdownEmits)

let timer: ReturnType<typeof setInterval> | undefined

const rawValue = ref(getTime(props.value) - Date.now())

const displayValue = computed(() => formatTime(rawValue.value, props.format))

const formatter = (val: number) => formatTime(val, props.format)

const stopTimer = () => {
  if (timer) {
    cAF(timer)
    timer = undefined
  }
}

const startTimer = () => {
  const timestamp = getTime(props.value)

  const frameFunc = () => {
    let diff = timestamp - Date.now()
    emit('change', diff)
    if (diff <= 0) {
      diff = 0
      stopTimer()
      emit('finish')
    } else {
      timer = rAF(frameFunc)
    }
    rawValue.value = diff
  }
  timer = rAF(frameFunc)
}
```

I think use rAF is better.

Reply:
ok I'll change it later.